web_widget_x2many_2d_matrix/static/src/js/abstract_view_matrix_limit_extend.js

```diff
@@ -0,0 +1,22 @@
+odoo.define('web_widget_x2many_2d_matrix.matrix_limit_extend', function (require) {
+    "use strict";
+
+var AbstractView = require('web.AbstractView');
+
+var AbstractViewCustom = AbstractView.include({
+    // We extend this method so that the view is not limited to
+    // just 40 cells when the 'x2many_2d_matrix' widget is used.
+    _setSubViewLimit: function (attrs) {
+        this._super(attrs);
+        var view = attrs.views && attrs.views[attrs.mode];
+        var limit = view && view.arch.attrs.limit && parseInt(view.arch.attrs.limit, 0);
+        if (!limit && attrs.widget === 'x2many_2d_matrix') {
```

Comment:
AFAICS [here](https://github.com/odoo/odoo/blob/15c81d74a34b57c0aa0400ec8b66bf3397904c60/addons/web/static/src/js/views/abstract_view.js#L276), this condition will never be true, because the parent method always sets the limit, so basically this method will never do nothing; isn't it? :thinking: 

I think the implementation should be just:

```js
if (attrs.widget === 'x2many_2d_matrix') {
    var view = attrs.views && attrs.views[attrs.mode];
    _.defaults(view.arch.attrs, {limit: Infinity});
}
this._super(attrs);
```